test/server/cards/02_SHD/units/CriminalMuscle.spec.ts

```diff
@@ -0,0 +1,63 @@
+describe('Criminal Muscle', function () {
+    integration(function (contextRef) {
+        describe('Criminal Muscle\'s ability', function() {
+            it('should return an non-unique upgrade to owner\'s hand', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['frozen-in-carbonite'],
+                        groundArena: [{ card: 'luke-skywalker#jedi-knight', upgrades: ['lukes-lightsaber', 'jedi-lightsaber'] }],
+                        discard: ['entrenched']
+                    },
+                    player2: {
+                        hand: ['criminal-muscle'],
+                        groundArena: ['pyke-sentinel'],
+                        discard: ['hotshot-dl44-blaster']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.frozenInCarbonite);
+                context.player1.clickCard(context.pykeSentinel);
```

Comment:
why did you play frozenInCarbonite and you don't attach it from setup ?

Reply:
yeah, I didn't see a way on setup to designate who is the owner of the upgrade. I was wanting to make sure that the upgrade gets returned to the proper owners hand. Since I am new, this might be me over-testing a bit and should just trust the system.

Reply:
I didn't realize you want to test ownership of upgrade, I don't think you can set this property from setup

Reply:
@VJubert do you think it is overkill to test the ownership? Since I am unfamilar with the whole codebase, I did want to make too many assumptions in the tests but also want to respect the limited GitHub Actions runs the repo has.

Reply:
Doing more tests are never bad ^^